code/game/objects/items/weapons/RSF.dm

```diff
@@ -82,3 +82,75 @@ RSF
 	playsound(loc, 'sound/machines/click.ogg', 10, 1)
 	var/type_path = configured_items[mode][3]
 	new type_path(spawn_location)
+
+/obj/item/cookiesynth
+	name = "\improper Cookie Synthesizer"
+	desc = "A self-recharging device used to rapidly deploy cookies."
+	icon = 'icons/obj/tools.dmi'
+	icon_state = "rcd"
+	var/matter = 10
+	var/toxin = FALSE
+	var/cooldown = 0
+	var/cooldowndelay = 10
+	w_class = WEIGHT_CLASS_NORMAL
+
+/obj/item/cookiesynth/examine(mob/user)
+	. = ..()
+	to_chat(user, "<span class='notice'>It currently holds [matter]/10 cookie-units.</span>")
+
+/obj/item/cookiesynth/attackby()
+	return
+
+/obj/item/cookiesynth/emag_act(mob/user)
+	emagged = !emagged
+	if(emagged)
+		to_chat(user, "<span class='warning'>You short out [src]'s reagent safety checker!</span>")
+	else
+		to_chat(user, "<span class='warning'>You reset [src]'s reagent safety checker!</span>")
+		toxin = FALSE
+
+/obj/item/cookiesynth/attack_self(mob/user)
+	var/mob/living/silicon/robot/P = null
+	if(isrobot(user))
+		P = user
+	if(emagged && !toxin)
+		toxin = TRUE
+		to_chat(user, "<span class='warning'>Cookie Synthesizer Hacked.</span>")
+	else if(P.emagged && !toxin)
+		toxin = TRUE
+		to_chat(user, "<span class='warning'>Cookie Synthesizer Hacked.</span>")
+	else
+		toxin = FALSE
+		to_chat(user, "<span class='notice'>Cookie Synthesizer Reset.</span>")
+
+/obj/item/cookiesynth/process()
+	if(matter < 10)
+		matter++
+
+/obj/item/cookiesynth/afterattack(atom/A, mob/user, proximity)
+	if(cooldown > world.time)
+		return
+	if(!proximity)
+		return
+	if(!(istype(A, /obj/structure/table) || isfloorturf(A)))
+		return
+	if(matter < 1)
+		to_chat(user, "<span class='warning'>[src] doesn't have enough matter left. Wait for it to recharge!</span>")
+		return
+	if(isrobot(user))
+		var/mob/living/silicon/robot/R = user
+		if(!R.cell || R.cell.charge < 400)
+			to_chat(user, "<span class='warning'>You do not have enough power to use [src].</span>")
+			return
+	var/turf/T = get_turf(A)
+	playsound(loc, 'sound/machines/click.ogg', 10, 1)
+	to_chat(user, "Fabricating Cookie..")
```

Comment:
This one's just a nitpick, but since it fabricates instantly this might work better as past-tense rather than present.
```suggestion
	to_chat(user, "<span class='notice'>Cookie fabricated!</span>")
```